app/database/scripts/wellsearch/wells_replication_stored_functions.sql

```diff
@@ -969,29 +969,36 @@ BEGIN
 
   -- These next 3 lines will be removed once the new Aquifers app is completed (CodeWithUs)
   raise notice '...gw_aquifer_well data imported';
-  SELECT count(*) from aquifer_well into row_count;
+  SELECT count(*) FROM aquifer_well INTO row_count;
   raise notice '% rows loaded into the aquifer_well table',  row_count;
 
   raise notice '...gw_aquifer_attrs data imported';
-  SELECT count(*) from aquifer into row_count;
+  SELECT count(*) FROM aquifer INTO row_count;
   raise notice '% rows loaded into the aquifer table',  row_count;
 
   UPDATE aquifer
   SET    mapping_year = xform.mapping_year
   FROM   xform_aquifers xform
   WHERE  aquifer.aquifer_id = xform.aquifer_id
-  AND    xform.mapping_year is not null;
+  AND    xform.mapping_year IS NOT NULL;
 
   raise notice '...aquifer data updated from mapping spreadsheet';
 
+-- With multiple aquifers per well, we choose the 'last updated' one 
+  WITH best_match AS (
+    SELECT DISTINCT ON (well.well_id)
+    well.well_tag_number,aws.aquifer_id  
+    FROM  wells.gw_aquifer_wells aws,
+          wells.wells_wells well
+    WHERE well.well_id = aws.well_id
+    ORDER BY well.well_id, coalesce(aws.when_updated, aws.when_created) DESC  
```

Comment:
Do all the records have either a when_updated or when_created? If for whatever reasons they're null anywhere, we could always order by the gw_aquifer_wells id to try and get the latest - trusting that the id's are created in order. Of course, if they date values are always populated this isn't a concern.

Reply:
When_created is guaranteed to have a value, hence my use of the coalesce() function on the order by.

Reply:
See my other comment about when_created bring mandatory. Oh also note that the aquifer_well table will be removed once @rfb 's code is merged in.